example/detect-video.cpp

```diff
@@ -0,0 +1,98 @@
+
+    if (!cap.isOpened()) {
+        fprintf(stderr, "Failed to load the video file.");
+        return -1;
+    }
+
+    Mat im;
+    int* pResults = NULL;
+    //pBuffer is used in the detection functions.
+    //If you call functions in multiple threads, please create one buffer for each thread!
+    unsigned char* pBuffer = (unsigned char*)malloc(DETECT_BUFFER_SIZE);
+    if (!pBuffer)
+    {
+        fprintf(stderr, "Can not alloc buffer.\n");
+        return -1;
+    }
+
+    while(true)
+    {
+        cap >> im;
+
+        Mat image = im.clone();
+
+        Size origin(im.cols, im.rows);
+
+        Size size(320, 240);
+
+        resize(im, image, size);//resize image
+
+        ///////////////////////////////////////////
+        // CNN face detection 
+        // Best detection rate
+        //////////////////////////////////////////
+        //!!! The input image must be a BGR one (three-channel) instead of RGB
+        //!!! DO NOT RELEASE pResults !!!
+
+        pResults = facedetect_cnn(pBuffer, (unsigned char*)(image.ptr(0)), image.cols, image.rows, (int)image.step);
+
+        Mat result_image = image.clone();
+        //print the detection results
+        for (int i = 0; i < (pResults ? *pResults : 0); i++)
+        {
+            short* p = ((short*)(pResults + 1)) + 142 * i;
+            int confidence = p[0];
+            int x = p[1];
+            int y = p[2];
+            int w = p[3];
+            int h = p[4];
+             
+            //show the score of the face. Its range is [0-100]
+            char sScore[256];
+            snprintf(sScore, 256, "%d", confidence);
+            cv::putText(result_image, sScore, cv::Point(x, y - 3), cv::FONT_HERSHEY_SIMPLEX, 0.5, cv::Scalar(0, 255, 0), 1);
+
+            //draw face rectangle
+            rectangle(result_image, Rect(x, y, w, h), Scalar(0, 255, 0), 2);
+            //draw five face landmarks in different colors
+            cv::circle(result_image, cv::Point(p[5], p[5 + 1]), 1, cv::Scalar(255, 0, 0), 2);
+            cv::circle(result_image, cv::Point(p[5 + 2], p[5 + 3]), 1, cv::Scalar(0, 0, 255), 2);
+            cv::circle(result_image, cv::Point(p[5 + 4], p[5 + 5]), 1, cv::Scalar(0, 255, 0), 2);
+            cv::circle(result_image, cv::Point(p[5 + 6], p[5 + 7]), 1, cv::Scalar(255, 0, 255), 2);
+            cv::circle(result_image, cv::Point(p[5 + 8], p[5 + 9]), 1, cv::Scalar(0, 255, 255), 2);
+
+            //print the result
+            printf("face %d: confidence=%d, [%d, %d, %d, %d] (%d,%d) (%d,%d) (%d,%d) (%d,%d) (%d,%d)\n",
+                    i, confidence, x, y, w, h,
+                    p[5], p[6], p[7], p[8], p[9], p[10], p[11], p[12], p[13], p[14]);
+            
+        }
+
+        resize(result_image, result_image, origin);
+
+        // press 'q' to stop the program
+        cv::imshow("VideoDetect", result_image);
+        if ((cv::waitKey(2) & 0xFF) == 'q') break;
+    }
+
+//release the buffer
+free(pBuffer);
+
+return 0;
+}
```

Comment:
There is no reason to add this new cpp file for detecting faces in a video. `examples/detect-camera.cpp` can already do this.